pkg/catalog/interrogate.go

```diff
@@ -244,20 +244,36 @@ func AppCapabilities(
 	return appCR.Spec.Capabilities, nil
 }
 
-// AppPersistDirs fetches the required directories for a given app that
-// has be persisted on a PVC
+// AppPersistDirs fetches the required directories for a given role that
+// has be persisted on a PVC. If the role doesn't have an explicit list, use
+// the top level list.
 func AppPersistDirs(
 	cr *kdv1.KubeDirectorCluster,
+	role string,
 ) ([]string, error) {
 
+	var appPersistDirs []string
 	// Fetch the app type definition if we haven't yet cached it in this
 	// handler pass.
 	appCR, err := GetApp(cr)
 	if err != nil {
 		return []string{}, err
 	}
 
-	return appCR.Spec.PersistDirs, nil
+	// Check to see if there is a role-specific setup package.
+	for _, nodeRole := range appCR.Spec.NodeRoles {
+		if nodeRole.ID == role {
+			appPersistDirs = nodeRole.PersistDirs
+			break
+		}
+	}
+
+	// If role-specific persist_dirs is not present, use the main one.
+	if len(appPersistDirs) == 0 {
```

Comment:
The only thing that bugs me a little about this approach is that it won't be possible to specify an empty-list persist_dirs at the role level (to override some other value from the top). Actually we have the same issue with setup_package.

To allow that sort of behavior I think we'd need to do the "pointer trick" like with Role.Members, so that we can distinguish "didn't specify anything" from "specified emptylist" (or in the case of setup_package, "specified emptystring").

Hm. Thoughts?

Reply:
make sense. will do